codec/proto_codec.go

```diff
@@ -43,6 +43,10 @@ func NewProtoCodec(interfaceRegistry types.InterfaceRegistry) *ProtoCodec {
 // NOTE: this function must be used with a concrete type which
 // implements proto.Message. For interface please use the codec.MarshalInterface
 func (pc *ProtoCodec) Marshal(o ProtoMarshaler) ([]byte, error) {
+	// check nil messages early on to avoid panic.
+	if o.Size() == 0 {
```

Comment:
```suggestion
	if o == nil || o.Size() == 0 {
```

Reply:
`o == nil` seems useless here because it can't catch the nil value.

Reply:
thanks for the PR, any chance we can have a simple test to verify the previous panic behaviour and that it is fixed with this.

Reply:
Yes, let's please add a test and changelog entry.

Also, I don't understand the `nil` comment @yihuang. What do you mean you can't catch the nil value? I see no charm in adding a nil check.

Reply:
- the `nil` check is added
- unit test added, it would fail without the new changes.